File: ruby/Strategy/tsv_extraction.rb
require_relative 'extraction'

class TSVExtraction < Extraction

  def initialize
  end

  def extract
    open
    print "I will extract by seperating row by tabs\n"
    close
  end

  private
  def open
    print "Opening TSV file\n"
  end

  private
  def close
    print "Closing TSV file\n"
  end
end

File: java/Strategy/src/TSVExtraction.java

class TSVExtraction implements Extraction {

	public void extract() {
		// complete the method below
		
		open();
		System.out.println("I will extract by separating row by tabs.");
		close();
	}

	public void open() {
		System.out.println("Opening TSV file");
	}

	public void close() {

		System.out.println("Closing TSV file");
	}
}